File: oim-jruby/update_users.rb
require 'java'
require 'xlclient'

include_class('java.lang.Exception') {|package,name| "J#{name}" }
include_class 'java.lang.System' 
include_class 'java.util.HashMap'
include_class('Thor.API.tcUtilityFactory') {|package,name| "OIM#{name}"}


numUsers = 100
prefix = 'hundred'

xlclient = XLAPIClient.new
xlclient.defaultLogin

usrIntf = xlclient.getUtility('usr')

for i in (1..numUsers)
    usrHash = {
        'Users.User ID' => prefix + i.to_s,
        'Users.First Name' => prefix + 'First' + i.to_s,
        'Users.Last Name' => prefix + 'Last' + i.to_s,
        'Users.Middle Name' => prefix + 'Middle' + i.to_s,
        'Users.Password' => 'foo',
        'Users.Role' => 'Full-Time',
        'Users.Xellerate Type' => 'End-User',
        'Organizations.Key' => '1'
    }

    usrMap = HashMap.new(usrHash)

    t1 = System.currentTimeMillis
    usrKey = usrIntf.createUser(usrMap)
	t2 = System.currentTimeMillis

    delta = t2-t1
    puts "Created user with key = #{usrKey} time = #{delta}"
end

xlclient.close
System.exit 0

File: oim-jython/updateUsers.py
from xlclient import *
from java.util import *
from java.lang import *

import sys

if __name__ == '__main__':
    xlclient = XLAPIClient('xelsysadm', 'xelsysadm')
    usrIntf = xlclient.usrIntf
    usrCols = ['Users.Key', 'Users.Row Version', 'Users.User ID']


    usrMap = HashMap()


    for i in range(4, 110009):	
	usrMap.put('Users.Key', str(i))

	rs = usrIntf.findUsersFiltered(usrMap, usrCols)

	updateMap = HashMap()
	updateMap.put('Users.First Name', rs.getStringValue('Users.User ID') + 'Fst')
	updateMap.put('Users.Last Name', rs.getStringValue('Users.User ID') + 'Fst')
	t1 = System.currentTimeMillis()
	usrIntf.updateUser(rs, updateMap)
	t2 = System.currentTimeMillis()
	delta = t2-t1
	
	print 'Updating user ', rs.getStringValue('Users.User ID'), ' time = ', str(delta)

    xlclient.close()
    sys.exit(0)
